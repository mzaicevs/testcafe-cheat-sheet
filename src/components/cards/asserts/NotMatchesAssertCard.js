import React from "react";
import { assert_notMatch } from "../../../snippets";
import CodeCard from "../../../base_components/CodeCard";

const NotMatchesAssertCard = () => (
  <CodeCard
    cardTitle={assert_notMatch.title}
    fullDocLink={assert_notMatch.fullDocLink}
    codeSnippet={{
      code: assert_notMatch.code,
      language: assert_notMatch.language,
      options: assert_notMatch.options,
      example: assert_notMatch.example
    }}
    description={assert_notMatch.description}
  />
);

export default NotMatchesAssertCard;
