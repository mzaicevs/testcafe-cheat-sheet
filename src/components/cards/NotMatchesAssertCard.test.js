import React from "react";
import { shallow } from "enzyme";
import NotMatchesAssertCard from "./NotMatchesAssertCard";

jest.mock("react-syntax-highlighter/dist/esm/styles/hljs", () => () => {});

describe("NotMatchesAssertCard", () => {
  it("matches snapshot", () => {
    const wrapper = shallow(<NotMatchesAssertCard />);
    expect(wrapper).toMatchSnapshot();
  });
});
