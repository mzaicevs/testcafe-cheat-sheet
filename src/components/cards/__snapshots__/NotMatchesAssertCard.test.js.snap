// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotMatchesAssertCard matches snapshot 1`] = `
<CodeCard
  cardTitle="Not Match - Assert"
  codeSnippet={
    Object {
      "code": Array [
        "t.expect( actual ).notMatch( re, message, options );",
      ],
      "example": "import { ClientFunction } from 'testcafe';

  fixture \`My fixture\`;
  
  test('My test', async t => {
      const getLocation = ClientFunction(() => document.location.href.toString());
      
      await t
        .expect(getLocation()).notMatch(/.ie/)
        .expect('foobar').notMatch(/^b/, 'this assertion will be passed');
  });",
      "language": "javascript",
      "options": "{ 
    timeout: Number,
      allowUnawaitedPromise: Boolean 
  }",
    }
  }
  description="Asserts that actual does not matches the 're' regular expression."
  fullDocLink="https://devexpress.github.io/testcafe/documentation/test-api/assertions/assertion-api.html#not-match"
/>
`;
